mantidproject/mantid
# New adjustable collapsiblestack for the instrument viewer

### Description of work

Improved the collapsible stack from the instrument viewer pick tab (the widget that holds the plot, info, and rebin widget). The main difference is that panels are held within a QSplitter so can be resized by clicking and dragging. Also fixed some weird behaviour if you only had the rebin widget expanded.

#### Summary of work

New files `CollapsibleStack.h/cpp` to replace `CollapsiblePanel.h/cpp`. Public function names and class have stayed the same, so minimal code changes are needed elsewhere. When adding a panel you can specify `fixedHeight=true` to lock the widget size in the qsplitter, this is done for the rebin widget.

Fixes #32118

### To test:

- Load some data and open in instrument view.
- Go to the pick tab.
- Play around with the collapsible stack, moving the splitter partition, showing and hiding different combinations of the widgets to check there is no weird behaviour.
- Make sure it is working the same in ALFView.



---

### Reviewer

Please comment on the points listed below ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)).
**Your comments will be used as part of the gatekeeper process, so please comment clearly on what you have checked during your review.** If changes are made to the PR during the review process then your final comment will be the most important for gatekeepers. In this comment you should make it clear why any earlier review is still valid, or confirm that all requested changes have been addressed.

#### Code Review

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Do the release notes conform to the [release notes guide](https://developer.mantidproject.org/Standards/ReleaseNotesGuide.html)?

#### Functional Tests

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### Gatekeeper

If you need to request changes to a PR then please add a comment and set the review status to "Request changes". This will stop the PR from showing up in the list for other gatekeepers.
